swarm/src/handler.rs

```diff
@@ -125,25 +124,17 @@ pub trait ConnectionHandler: Send + 'static {
 
     /// Returns until when the connection should be kept alive.
     ///
-    /// This method is called by the `Swarm` after each invocation of
-    /// [`ConnectionHandler::poll`] to determine if the connection and the associated
-    /// [`ConnectionHandler`]s should be kept alive as far as this handler is concerned
-    /// and if so, for how long.
-    ///
-    /// Returning [`KeepAlive::No`] indicates that the connection should be
-    /// closed and this handler destroyed immediately.
-    ///
-    /// Returning [`KeepAlive::Until`] indicates that the connection may be closed
-    /// and this handler destroyed after the specified `Instant`.
-    ///
-    /// Returning [`KeepAlive::Yes`] indicates that the connection should
-    /// be kept alive until the next call to this method.
+    /// This method is an optional implementation and can be called by the `Swarm` after
+    /// each invocation of [`ConnectionHandler::poll`] to determine if the connection
+    /// and the associated [`ConnectionHandler`]s should be kept alive.
```

Comment:
We should bring back the docs we had here earlier that mention the keep alive algorithm based on stream counting.

We should also mention:

- How users can opt out of it (`Stream::no_keep_alive`)
- That this is only called once the `ConnectionHandler` is idle (i.e. returns `Poll::Pending`) AND we don't have active stream upgrades AND we don't have active streams.

In a nutshell, we should just recommend users that they usually don't need to override this unless they want their connection to be kept alive _beyond_ any activity on it.

Reply:
updated 3d10eb4